app/services/user_search.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class UserSearch
+  include ServicePattern
+
+  class Result
+    attr_reader :users, :limit
+
+    def initialize(users:, limit:)
+      @users = users
+      @limit = limit
+    end
+  end
+
+  MIN_QUERY_LENGTH = 2
+  DEFAULT_LIMIT = 15
+
+  def initialize(query: nil, limit: DEFAULT_LIMIT)
+    @query = ReplaceAbbreviation.call(string: StripPunctuation.call(string: query))
+    @limit = limit
+  end
+
+  def call
+    Result.new(users: specified_users, limit: limit)
+  end
+
+  def specified_users
+    users = User.all
+    users = users.search(query) if query
+    users = users.limit(limit) if limit
+    users
```

Comment:
Hmmm I wonder whether we can apply some kind of sort here, or what it feels like without a sort with out full dataset..